brave/brave-ios
# Fix #8349: Transaction Activity v2 UI

## Summary of Changes
- Integrates v2 UI for Activity tab, including search/filter transactions. Whole lot of UI!
- Created a new `TransactionSummaryViewContainer` that can be passed a `ParsedTransaction` for display. Different subview used for each tx type as appropriate.
- Created a `TransactionsListView` to contain the UI. The intention is for this to be re-usable for Asset Details v2 & Account Details transaction list. Activity tab has a wrapper around this view.
- Some smaller changes / improvements:
    - ERC20 Approve transaction fiat now calculated in `ParsedTransaction`
    - Added `mini` size to `LoaderView` / brave circlular loading indicator
    - `func localImage(network: BraveWallet.NetworkInfo) -> UIImage?` moved to `BlockchainToken` extension for re-use.
    - Added `NetworkInfo` to `ParsedTransaction` (previously only stored network symbol.

This pull request fixes #8349

## Submitter Checklist:


## Test Plan:
- Verify all transaction types show up as expected, should be similar to desktop Nightly. Send on EVM, send on Solana, send on Filecoin, Swap on EVM, Swap on Solana (no icon information available to show), ERC20 Approval, dapp transactions.
- Verify transaction types are grouped by the day (date header vs time headers)
- Verify search / filtering works. Search supports account names, addresses, token name, token symbol, token contract address and transaction hash.


## Screenshots:

![activity empty - light](https://github.com/brave/brave-ios/assets/5314553/7d445682-4178-444c-80bb-732b9be12fea) | ![activity empty - dark](https://github.com/brave/brave-ios/assets/5314553/76a7dc49-7afa-4296-8501-1a361c6b8e21)
--|--
![activity - light](https://github.com/brave/brave-ios/assets/5314553/79284170-1c1c-4fa6-b985-692695b5488f) | ![activity - dark](https://github.com/brave/brave-ios/assets/5314553/41237521-15fc-4ddc-b545-76e4d38aa5b3)


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
